### ISSUE TITLE
Setup npm releases

### ISSUE THREAD

**Issue by wotzlaff** [romgrk/node-gtk#226]
This would make the library a lot more usable. There seems to be some (older) script handling publishing.
Is it possible to implement some procedure in order to publish releases to npm on a regular basis?

**Comment by romgrk** [comment]
Yes, it's been way too long since the last release and many bugs have been fixed. In [docs/maintainers.md](https://github.com/romgrk/node-gtk/blob/master/doc/maintainers.md) the release procedure is documented. Basically it's: `npm version xxxx` && `npm publish`, then push a commit with `[publish binary]` in it and Travis will generate pre-built binaries for all the combinations of environments that we support, and upload them to an S3 bucket.

What do you have in mind by on a regular basis? A script that would automatically publish releases, eg when master is updated?

**Comment by romgrk** [comment]
Oh yes I remember why I haven't release in the last months: #217 

The macOS builds are failling and that will prevent Travis from generating the prebuilt binaries. TBH I've been putting that task off. The problem is on macOS, Cairo pulls some system header that defines `Boolean`, and that conflicts with `v8::Boolean` :/ I need to find the last working version of Cairo and check what changed. I also opened an issue on their tracker but haven't received feedback yet.

**Comment by wotzlaff** [comment]
I think we should not publish every tiny step (i.e. each commit to master). But at least every now and then a working version should be published.

Too bad that these macOS builds are failing again. I would like to help but I have no real macOS system available.
I would also love to see a working Windows build. I might try to get it working if I have some time.

BTW: Do you have any idea why travis isn't running anymore? I did set it up now for my fork to get the nice build and test logs...

**Comment by WebReflection** [comment]
if there's some test needed, I have a dusty mac mini around updated to latest macOS ... it might become useful again :-)

**Comment by romgrk** [comment]
@wotzlaff agreed. If you have any formal suggestion I'd be happy to hear it.
 - macOS: yes I'd really like to be cross-platform :/
 - Windows: same. The ground work had been done by the previous maintainer but there's probably a bit of work to make it work. At least I've already seen PyGObject run on Windows so I trust that it's possible.
 - Travis: [he does run](https://travis-ci.org/github/romgrk/node-gtk), but for some reason it doesn't show up on github. I've been checking the results directly on travis-ci.org but it's inconvenient.

@WebReflection lol well it's more like debugging a build problem on macOS. If you're willing to dig into it I'd be more than happy to have your assistance, but I'd understand that you're not. Anyway I have travis's logs and I can debug the build on a branch, but obviously it's way less practical than having a shell.

**Comment by WebReflection** [comment]
@romgrk I'm not familiar with building anything on macOS but if there's a procedure I can follow I can give it a try and see where I go

**Comment by romgrk** [comment]
Fyi I've released v0.5.0 just now even if the macOS builds are broken, v0.4.0 contains many issues that have been fixed so I'd rather have it out now.

And for the macOS builds related comments, please continue the discussion in #217 :)
